server/fleet/teams.go

```diff
@@ -143,6 +143,28 @@ func ValidGlobalRoles() []string {
 	return roles
 }
 
+// ValidateRole returns nil if the global and team roles combination is a valid
+// one within fleet, or a fleet Error otherwise.
+func ValidateRole(globalRole *string, teamUsers []UserTeam) error {
+	if globalRole == nil || *globalRole == "" {
+		if teamUsers == nil || len(teamUsers) == 0 {
```

Comment:
Nit: I prefer the simpler check allowing `len` to handle `nil` values. If you prefer the explicit `nil` check I'm okay leaving it as-is.

```suggestion
		if len(teamUsers) == 0 {
```